### PR TITLE
Destructured assignment variable names

### PR DESCRIPTION
**Pull request by jacknoble**
https://github.com/lpil/dogma/issues/89

### CODE REVIEW

File: test/dogma/rules/variable_name_test.exs [hunk 1]
```diff
@@ -38,13 +38,115 @@ defmodule Dogma.Rules.VariableNameTest do
...
+      errors = """
+      %{test: fooBar} = foo_bar
+      """ |> test
+      expected_errors = [
+        %Error{
+          rule:     VariableName,
+          message:  "Variable names should be in snake_case",
+          line: 2,
+        },]
+      assert expected_errors = errors
```
Comment (lpil) [inline test/dogma/rules/variable_name_test.exs]: `==` instead of `=` here?

**Comment by lpil** [comment]
This is great! Was really cool to see the tests handling multiple levels of nested data structures

File: test/dogma/rules/variable_name_test.exs (hunk 1 above)
Comment (jacknoble) [inline test/dogma/rules/variable_name_test.exs]: whoops!

**Comment by jacknoble** [comment]
Added binary matching and head/tail matching. Anything missing?

**Comment by lpil** [comment]
That's really cool. Thank you!

**Pull request merged**